RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -6,21 +6,42 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.document.Document;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.instrumentation.annotations.SpanAttribute;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 import reactor.util.retry.Retry;
 
 
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
+    private static final LongCounter documentsProcessedCounter;
+    private static final LongCounter batchesProcessedCounter;
+    static {
+        final var meter = GlobalOpenTelemetry.getMeter("DocumentReindexer");
+        documentsProcessedCounter = meter.counterBuilder("documentsProcessed")
+            .setDescription("Counts the number of documents processed")
+            .setUnit("1")
+            .build();
+        batchesProcessedCounter = meter.counterBuilder("batchesProcessed")
+            .setDescription("Counts the number of batches processed")
+            .setUnit("1")
+            .build();
+    }
+
     private static final int MAX_BATCH_SIZE = 1000; // Arbitrarily chosen
 
-    public static Mono<Void> reindex(String indexName, Flux<Document> documentStream, OpenSearchClient client) throws Exception {   
+    @WithSpan
+    public static Mono<Void> reindex(@SpanAttribute("indexName") String indexName, Flux<Document> documentStream, OpenSearchClient client) throws Exception {   
 
         return documentStream
             .map(DocumentReindexer::convertDocumentToBulkSection)  // Convert each Document to part of a bulk operation
+            .map(d -> { documentsProcessedCounter.add(1); return d; })
             .buffer(MAX_BATCH_SIZE) // Collect until you hit the batch size
-            .doOnNext(bulk -> logger.info(bulk.size() + " documents in current bulk request"))
+            .doOnNext(bulk -> { logger.info(bulk.size() + " documents in current bulk request"); batchesProcessedCounter.add(1); })
```

Comment:
two lines for the different statements please.

Reply:
ACK - this was a quick hack that should never get past draft PRs